### ISSUE TITLE
outfit with proper logging

### ISSUE THREAD

**Issue by eads** [newsdev/elex#23]

**Comment by jeremyjbowers** [comment]
Needs logging to watch progress tonight. Bumping to high.

**Comment by jeremyjbowers** [comment]
Should use the Python logging handler. Two routes at least; log to file and log to SysLogHandler for use with Papertrail for @cjdd3b.

**Comment by jeremyjbowers** [comment]
Dropping back to low. Handling this in the Django app for tonight.

**Comment by eads** [comment]
I moved this to the backlog. It's a tricky problem, it turns out.

All I really want is to log the URL and status code from the response object associated with an API request. It's nice to know exactly the generated URL for troubleshooting reasons.

Some solutions I examined and their downsides:

* Standard Python logger: Works with many systems, but not captured by Cement logger and also generally a bit of a pain in the ass to keep log message format consistent, etc.
* Use Cement's `minimal_logger`: Captured by Cement, but hard-codes console output and argument handling; I didn't test but it also seems unlikely to be captured by other logging implementations/configurations (e.g. Django).
* Refactoring the API request pipeline to return a response object and set on Election instances: This is more or less the right solution I think. Any commands calling the Elex python API can simply log as they see fit using models (and their properties) from the Elex Python API. Where this breaks down is getting elections and next elections. Easy enough to set a _response or _last_response property on an Election instance. But the stuff that grabs data about elections themselves are class methods, not instance methods.

We can get most of the way there by following this last path if we accept that we don't have a solution for the class methods yet and ticket that out. Or we try to do some fancier acrobatics to route logging in to the python api to  the cement log handler. I think the first option is the best plan for now. 

In all events, I don't think this needs to block any release for now --by and large the CLI logging works great.
